pcdsutils/import_timer.py

```diff
@@ -0,0 +1,370 @@
+    ----------
+    root_module : str
+        The source module, e.g. pcdsdevices
+    self_time_raw : int
+        The time it took to import this module in microseconds, not counting
+        the time it took to import other modules.
+    self_time : float
+        The time it took to import this module in seconds, not counting
+        the time it took to import other modules.
+    cumulative_time_raw : int
+        The time it took to import this module in microseconds,
+        including the time it took to import other modules.
+    cumulative_time : float
+        The time it took to import this module in seconds,
+        including the time it took to import other modules.
+    submodule_stats : tuple of ImportTimeStats
+        The stats that were used to generate this stats summary.
+    """
+    root_module: str
+    self_time_raw: int
+    self_time: float
+    cumulative_time_raw: int
+    cumulative_time: float
+    submodule_stats: Tuple[ImportTimeStats]
+
+    @classmethod
+    def from_stats(cls, stats: List[ImportTimeStats]) -> ModuleStatsSummary:
+        """
+        Create a ModuleStatsSummary from a list of ImportTimeStats.
+        """
+        self_time_raw = 0
+        cumulative_time_raw = 0
+        for stat in stats:
+            self_time_raw += stat.self_time_raw
+            cumulative_time_raw = max(
+                cumulative_time_raw,
+                stat.cumulative_time_raw,
+            )
+        return cls(
+            root_module=stats[0].root_module,
+            self_time_raw=self_time_raw,
+            self_time=self_time_raw / 1e6,
+            cumulative_time_raw=cumulative_time_raw,
+            cumulative_time=cumulative_time_raw / 1e6,
+            submodule_stats=tuple(stats),
+        )
+
+    def show_detailed_summary(self, sort_key: str = 'self_time') -> None:
+        """
+        For this module, explain the situation to stdout.
+        """
+        field_names = tuple(
+            field for field in ImportTimeStats.__dataclass_fields__
+            if 'raw' not in field
+        )
+        table = PrettyTable(
+            field_names=field_names,
+        )
+        summary_row = list(
+            getattr(self, attr, 'N/A')
+            for attr in field_names
+        )
+        summary_row[0] = 'Total'
+        table.add_row(summary_row)
+        for stats in sorted(
+            self.submodule_stats,
+            key=lambda s: getattr(s, sort_key),
+            reverse=True,
+        ):
+            table.add_row(
+                tuple(
+                    getattr(stats, attr)
+                    for attr in field_names
+                )
+            )
+        print(table)
+
+
+def get_import_time_text(module: str) -> List[str]:
+    """
+    Run python -X importtime modulename in a subprocess.
+
+    Returns the line-by-line std output from this call.
+    """
+    return subprocess.check_output(
+        ['python', '-X', 'importtime', '-c', f'import {module}'],
```

Comment:
Perhaps
```suggestion
        [sys.executable, '-X', 'importtime', '-c', f'import {module}'],
```

Reply:
https://docs.python.org/3/library/sys.html#sys.executable
> A string giving the absolute path of the executable binary for the Python interpreter, on systems where this makes sense. If Python is unable to retrieve the real path to its executable, [sys.executable](https://docs.python.org/3/library/sys.html#sys.executable) will be an empty string or None.

So maybe I make this `sys.executable or 'python'`?